internal/filer_test.go

```diff
@@ -463,43 +458,97 @@ func TestAccFilerWorkspaceNotebookWithOverwriteFlag(t *testing.T) {
+			nameWithoutExt: "sqlJupyterNotebook",
+			content1:       readFile(t, "testdata/notebooks/sql1.ipynb"),
+			expected1:      "-- Databricks notebook source\nselect 1",
+			content2:       readFile(t, "testdata/notebooks/sql2.ipynb"),
+			expected2:      "-- Databricks notebook source\nselect 2",
+		},
+	} {
+		t.Run(tcases.name, func(t *testing.T) {
+			t.Parallel()
 
-	// Assert contents after initial upload
-	filerTest{t, f}.assertContents(ctx, "pyNb", "# Databricks notebook source\nprint('first upload'))")
-	filerTest{t, f}.assertContents(ctx, "rNb", "# Databricks notebook source\nprint('first upload'))")
-	filerTest{t, f}.assertContents(ctx, "sqlNb", "-- Databricks notebook source\n SELECT \"first upload\"")
-	filerTest{t, f}.assertContents(ctx, "scalaNb", "// Databricks notebook source\n println(\"first upload\"))")
-	filerTest{t, f}.assertContents(ctx, "jupyterNb", "# Databricks notebook source\nprint(\"Jupyter Notebook Version 1\")")
+			// Upload the notebook
+			err = f.Write(ctx, tcases.name, strings.NewReader(tcases.content1))
+			require.NoError(t, err)
 
-	// Upload notebooks a second time, overwriting the initial uplaods
-	err = f.Write(ctx, "pyNb.py", strings.NewReader("# Databricks notebook source\nprint('second upload'))"), filer.OverwriteIfExists)
-	require.NoError(t, err)
-	err = f.Write(ctx, "rNb.r", strings.NewReader("# Databricks notebook source\nprint('second upload'))"), filer.OverwriteIfExists)
-	require.NoError(t, err)
-	err = f.Write(ctx, "sqlNb.sql", strings.NewReader("-- Databricks notebook source\n SELECT \"second upload\""), filer.OverwriteIfExists)
-	require.NoError(t, err)
-	err = f.Write(ctx, "scalaNb.scala", strings.NewReader("// Databricks notebook source\n println(\"second upload\"))"), filer.OverwriteIfExists)
-	require.NoError(t, err)
-	err = f.Write(ctx, "jupyterNb.ipynb", strings.NewReader(jupyterNotebookContent2), filer.OverwriteIfExists)
-	require.NoError(t, err)
+			// Assert contents after initial upload
+			filerTest{t, f}.assertContents(ctx, tcases.nameWithoutExt, tcases.expected1)
+
+			// Upload the notebook a second time with overwrite flag
+			err = f.Write(ctx, tcases.name, strings.NewReader(tcases.content2), filer.OverwriteIfExists)
+			require.NoError(t, err)
 
-	// Assert contents have been overwritten
-	filerTest{t, f}.assertContents(ctx, "pyNb", "# Databricks notebook source\nprint('second upload'))")
-	filerTest{t, f}.assertContents(ctx, "rNb", "# Databricks notebook source\nprint('second upload'))")
-	filerTest{t, f}.assertContents(ctx, "sqlNb", "-- Databricks notebook source\n SELECT \"second upload\"")
-	filerTest{t, f}.assertContents(ctx, "scalaNb", "// Databricks notebook source\n println(\"second upload\"))")
-	filerTest{t, f}.assertContents(ctx, "jupyterNb", "# Databricks notebook source\nprint(\"Jupyter Notebook Version 2\")")
+			// Assert contents after second upload
+			filerTest{t, f}.assertContents(ctx, tcases.nameWithoutExt, tcases.expected2)
+		})
+	}
```

Comment:
For follow-up: this is pretty much a copy of the test before this one.

It should be possible to include the overwrite behavior in the previous one, e.g. first test that it errors if the overwrite flag isn't set, then test that it does error with the overwrite flag set.

Reply:
Done, combined the tests.